tests/basic/shared_nfsv4_test.go

```diff
@@ -0,0 +1,173 @@
+					Inst().S.EnableSchedulingOnNode(node)
+				}
+			})
+
+			Step("fail over nfs server, and make sure the pod on server gets restarted", func() {
+				oldServer, err := Inst().V.GetNodeForVolume(volume, defaultCommandTimeout, defaultCommandRetry)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("old nfs server %v [%v]", oldServer.SchedulerNodeName, oldServer.Addresses[0])
+				pods, err := core.Instance().GetPodsUsingPV(volume.ID)
+				Expect(err).NotTo(HaveOccurred())
+				var oldPodOnOldServer corev1.Pod
+				for _, pod := range pods {
+					if pod.Spec.NodeName == oldServer.Name {
+						oldPodOnOldServer = pod
+					}
+				}
+				// make sure there is a pod running on the old nfs server
+				Expect(&oldPodOnOldServer).NotTo(BeNil())
+				logrus.Infof("pod on old server %v, creation time %v", oldPodOnOldServer.Name, oldPodOnOldServer.CreationTimestamp)
+
+				timestampBeforeFailOver := time.Now()
+				err = Inst().V.StopDriver([]node.Node{*oldServer}, false, nil)
+				Expect(err).NotTo(HaveOccurred())
+				err = Inst().V.WaitDriverDownOnNode(*oldServer)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("stopped px on nfs server node %v [%v]", oldServer.SchedulerNodeName, oldServer.Addresses[0])
+
+				var newServer *node.Node
+
+				for i := 0; i < 10; i++ {
+					server, err := Inst().V.GetNodeForVolume(volume, defaultCommandTimeout, defaultCommandRetry)
+					Expect(err).NotTo(HaveOccurred())
+					if server.Id != oldServer.Id {
+						logrus.Infof("nfs server failed over, new nfs server is %s [%s]", server.SchedulerNodeName, server.Addresses[0])
+						newServer = server
+						break
+					}
+					time.Sleep(oneMinute)
+				}
+				// make sure nfs server failed over
+				Expect(newServer).NotTo(BeNil())
+				logrus.Infof("new nfs server is %v", newServer)
+
+				logrus.Infof("start px on old nfs server Id %v, Name %v", oldServer.Id, oldServer.Name)
+				Inst().V.StartDriver(*oldServer)
+				err = Inst().V.WaitDriverUpOnNode(*oldServer, Inst().DriverStartTimeout)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("px is up on old nfs server Id %v, Name %v", oldServer.Id, oldServer.Name)
+
+				// make sure the pods on both old and new server are restarted
+				pods, err = core.Instance().GetPodsUsingPV(volume.ID)
+				Expect(err).NotTo(HaveOccurred())
+				for _, pod := range pods {
+					if pod.Spec.NodeName == oldServer.Name {
+						logrus.Infof("pod on old server %v, creation time %v", oldPodOnOldServer.Name, oldPodOnOldServer.CreationTimestamp)
+						logrus.Infof("After failover, pod on old server %v, creation time %v", pod.Name, pod.CreationTimestamp)
+						Expect(pod.CreationTimestamp.After(timestampBeforeFailOver)).To(BeTrue())
+					}
+					if pod.Spec.NodeName == newServer.Name {
+						logrus.Infof("After failover, pod on new server %v, creation time %v", pod.Name, pod.CreationTimestamp)
+						Expect(pod.CreationTimestamp.After(timestampBeforeFailOver)).To(BeTrue())
+					}
+				}
```

Comment:
we should assert that this loop found 2 pods. 

Also, this logic works only for repl=2 volumes. For repl=3 volumes, the new NFS server could be on any one of the remaining 2 replica nodes. So, the second pod may not need to be restarted. Maybe assert that it is a repl=2 volume at the top of the test so we don't waste time debugging failures for repl=3 vols.